File: algorithms/926-flip-string-to-monotone-increasing/flip-string-to-monotone-increasing.go
package problem926

func minFlipsMonoIncr(s string) int {
	dp0, dp1 := 0, 0
	for _, c := range s {
		dp0New, dp1New := dp0, min(dp0, dp1)
		if c == '1' {
			dp0New += 1
		} else {
			dp1New += 1
		}
		dp0, dp1 = dp0New, dp1New
	}
	return min(dp0, dp1)
}

func min(x, y int) int {
	if x < y {
		return x
	}
	return y
}

File: algorithms/926-flip-string-to-monotone-increasing/flip-string-to-monotone-increasing.py
def min_flips_mono_incr(s: str) -> int:
    dp0, dp1 = 0, 0
    for c in s:
        dp0_new, dp1_new = dp0, min(dp0, dp1)
        if c == '1':
            dp0_new += 1
        else:
            dp1_new += 1
        dp0, dp1 = dp0_new, dp1_new
    return min(dp0, dp1)
